### PR TITLE
[M10-400] Implemented Ogury SDK Banners

### PR DESCRIPTION
**Pull request by carlosalcala-freestar**
**[M10-400](#)**

## Description, Motivation and Context

**What is the goal of ticket?  What are the acceptance criteria?**

Integrate Ogury SDK - Banners

## How Has This Been Tested? 

Manually tested App Build (in Progress)

## Does this change require updates outside of the code, like updating the documentation, adding a feature flag, or adding fields to a DB table? If so, was this done?

N/A

## Any tickets that this is dependent upon?

N/A

## Should reviewers manually test your changes?

No

## Evidence

### CODE REVIEW

**Review by maddapper — CHANGES_REQUESTED** [review]
Normally, we don't commit xcframework binaries into the source repo.   In fact, everything under the Pods folder should probably be git-ignroed.

File: FreestarAds-Ogury-PreRelease.local.podspec [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Pod::Spec.new do |spec|
+  spec.name                = "FreestarAds-Ogury-PreRelease"
+  spec.version             = "1.0.0.1-beta-1"
```
Comment (maddapper) [inline FreestarAds-Ogury-PreRelease.local.podspec]: Partner adapter semantic versioning schemes should not start at 1.0.0...

File: FreestarAds-Ogury-PreRelease.podspec [hunk 2]
```diff
@@ -0,0 +1,23 @@
+Pod::Spec.new do |spec|
+  spec.name                = "FreestarAds-Ogury-PreRelease"
+  spec.version             = "1.0.0.1-beta-1"
```
Comment (maddapper) [inline FreestarAds-Ogury-PreRelease.podspec]: Partner adapter semantic versioning schemes should not start at 1.0.0...

File: FreestarAds-Ogury.podspec [hunk 3]
```diff
@@ -0,0 +1,23 @@
+Pod::Spec.new do |spec|
+  spec.name                = "FreestarAds-Ogury"
+  spec.version             = "1.0.0-beta-1"
```
Comment (maddapper) [inline FreestarAds-Ogury.podspec]: Partner adapter semantic versioning schemes should not start at 1.0.0...

File: FreestarAds-Ogury/Source/objc/OguryBannerMediator.m [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+@end
+
+@implementation OguryBannerMediator
+
+- (BOOL)isEnabledForMediatorFormat:(FSTRMediatorFormatType)type {
+    switch (type) {
+        case FSTRMediatorFormatTypeBanner:
+            return YES;
+        default:
+            return NO;
```
Comment (maddapper) [inline FreestarAds-Ogury/Source/objc/OguryBannerMediator.m:33]: Does ogury support inline (mrec)? if so, we need to enable it here.

File: FreestarAds-Ogury/Source/objc/OguryBannerMediator.m [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+   [self partnerAdLoadFailed:[NSString stringWithFormat:@"%ld", (long)error.code]];
+}
+
+- (void)didTriggerImpressionOguryBannerAd:(OguryBannerAd *)banner {
+    FSTRLog(@"OGURY: didTriggerImpressionOguryBannerAd");
+}
+
+- (UIViewController *)presentingViewControllerForOguryAdsBannerAd:(OguryBannerAd*)banner {
+    return nil;
```
Comment (maddapper) [inline FreestarAds-Ogury/Source/objc/OguryBannerMediator.m]: Is this correct? I'm just wondering why we have a method that always returns nil.

File: FreestarAds-Ogury/Source/objc/OguryFSTRPartnerInitializer.m [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        instance = [[OguryFSTRPartnerInitializer alloc] init];
+    });
+    return instance;
+}
+
+- (void)runInitialization {
+    // run partner initialization here
+    FSTRLog(@"OGURY: Partner initialization.");
+    OguryConfigurationBuilder *configurationBuilder = [[OguryConfigurationBuilder alloc] initWithAssetKey:@"OGY-773308830772"];
+    [Ogury startWithConfiguration:[configurationBuilder build]];
```
Comment (maddapper) [inline FreestarAds-Ogury/Source/objc/OguryFSTRPartnerInitializer.m:28]: If you look at some of the other examples we have, the partner initializers need to inform back to the FSTRPartnerInitializerDelegate with initialization status.

**Review by maddapper — CHANGES_REQUESTED** [review]
@carlosalcala-freestar Please take a look at my review comments.

File: FreestarAds-Ogury-PreRelease.podspec (hunk 2 above)
Comment (carlosalcala-freestar) [reply FreestarAds-Ogury-PreRelease.podspec]: As per @kkawai comment this should be `2.1.0` right ? or what version should be ?

File: FreestarAds-Ogury/Source/objc/OguryBannerMediator.m (hunk 5 above)
Comment (carlosalcala-freestar) [reply FreestarAds-Ogury/Source/objc/OguryBannerMediator.m]: I will fix this @maddapper  it's the same problem we had on Yahoo adapter.

File: FreestarAds-Ogury/Source/objc/OguryFSTRPartnerInitializer.m (hunk 6 above)
Comment (carlosalcala-freestar) [reply FreestarAds-Ogury/Source/objc/OguryFSTRPartnerInitializer.m:28]: Will check

File: FreestarAds-Ogury/Source/objc/OguryBannerMediator.m (hunk 4 above)
Comment (carlosalcala-freestar) [reply FreestarAds-Ogury/Source/objc/OguryBannerMediator.m:33]: Sure will enable

**Review by maddapper — APPROVED** [review]

**Comment by carlosalcala-freestar** [comment]
Closing because Ogury Banners are already on PR https://github.com/freestarcapital/ios-sdk-ogury-adapter/pull/2

**Pull request closed without merging**